OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/cg/AndroidICCAnalysis.scala

```diff
@@ -1346,17 +1202,194 @@ class AndroidICCAnalysis(val project: SomeProject) extends FPCFAnalysis {
+                    val virtualCall = stmt.asAssignment.expr.asVirtualFunctionCall
+                    if (virtualCall.receiver.asVar.definedBy.head > -1)
+                        registeredReceivers += ObjectType(statements(virtualCall.
+                            receiver.asVar.definedBy.head).asAssignment.expr.asGetField.objRef.asVar.value.toString.
+                            replaceAll(receiverRegex, "").replaceAll("\\.", "/"))
+                }
+            }
+        }
+    }
+}
+
+object ManifestParsingKey extends ProjectInformationKey[(Map[String, ListBuffer[ClassFile]], ListBuffer[IntentFilter]), Elem] {
+
+    override def requirements(project: SomeProject): ProjectInformationKeys = Nil
+
+    override def compute(project: SomeProject): (Map[String, ListBuffer[ClassFile]], ListBuffer[IntentFilter]) = {
+        val manifest = project.getProjectInformationKeyInitializationData(ManifestParsingKey)
+        if (manifest.isDefined) {
+            parseManifest(project, manifest.get)
+        } else null
+    }
+
+    /**
+     * Analyses the AndroidManifest.xml of the project to find intent filters and relevant components to generate
+     * lifecycle callbacks.
+     */
+    def parseManifest(project: SomeProject, manifest: Elem): (Map[String, ListBuffer[ClassFile]], ListBuffer[IntentFilter]) = {
+        val componentMap: mutable.Map[String, ListBuffer[ClassFile]] = mutable.Map(
+            "activity" -> ListBuffer.empty[ClassFile],
+            "service" -> ListBuffer.empty[ClassFile]
+        )
+        val intentFilters = ListBuffer.empty[IntentFilter]
+        val androidURI = "http://schemas.android.com/apk/res/android"
+        val pack = manifest.attribute("package").get.toString().replaceAll("\\.", "/")
+        List("activity", "receiver", "service").foreach { comp ⇒
+            val components = manifest \\ comp
+            components.foreach { c ⇒
+                var ot = c.attribute(androidURI, "name").head.toString().replaceAll("\\.", "/")
+                if (ot.startsWith("/")) { ot = pack + ot }
+                val rec = project.classFile(ObjectType(ot))
+                if (rec.isDefined) {
+                    if (comp == "activity" || comp == "service") componentMap(comp) += rec.get
+                    val filters = c \ "intent-filter"
+                    if (filters.nonEmpty) {
+                        filters.foreach { filter ⇒
+                            val intentFilter = new IntentFilter()
+                            intentFilter.componentType = comp
+                            intentFilter.receiver = rec.get
+                            intentFilter.actions = (filter \ "action").map(_.attribute(androidURI, "name").
+                                get.head.toString()).to[ListBuffer]
+                            intentFilter.categories = (filter \ "category").map(_.attribute(androidURI, "name").
```

Comment:
You probably want to move all this initialization to be parameters of the IntentFilter class (i.e., constructor parameters).